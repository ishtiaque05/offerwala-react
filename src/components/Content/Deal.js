import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
// import { Link } from 'react-router-dom';

import { 
  withStyles, 
  Card, 
  CardActionArea, 
  CardContent, 
  CardMedia, 
  Typography, 
  Hidden
} from '@material-ui/core';

import DefaultImage from '../../assets/images/default_deal.jpg';
import DesktopDealDetails from './DealDetails/DesktopDealDetails';
import MobileDealDetails from './DealDetails/MobileDealDetails';

const styles = theme => ({
  card: {
    width: theme.spacing.unit * 43.75,
    maxHeight: theme.spacing.unit * 100,
    margin: theme.spacing.unit * 1.5,
    background: '#FFF',
    boxShadow: '0 4px 6px #D4D4D4',
    [theme.breakpoints.down('xs')]: {
      marginLeft: 'auto',
      marginRight: 'auto', 
      width: '100%'
    }
  },
  flex: {
    display: 'flex',
    justifyContent: 'space-between'
  },
  media: {
    height: 'auto'
  },
  link: {
    textDecoration: 'none'
  },
  shopName: {
    backgroundColor: '#4B4B4B',
    color: '#FFF',
    padding: theme.spacing.unit * 0.5,
    paddingLeft: theme.spacing.unit * 1.5,
    paddingRight: theme.spacing.unit * 1.5,
    borderRadius: theme.spacing.unit * 0.5
  },
  dealDetails: {
    margin: 0,
    padding: 0
  }
});

class Deal extends Component {
  state = {
    open: false
  };

  clickHandler = () => {
    this.setState({
      open: true
    });
  };

  handleClose = () => {
    this.setState({ open: false });
  };

  render() {
    const { classes, deal } = this.props;
    console.log(deal)

    return (
      <Fragment>

        <Hidden smUp>
          <MobileDealDetails 
            deal={deal}
            onClose={this.handleClose}
            open={this.state.open}
          /> 
        </Hidden>
        <Hidden smDown>
          <DesktopDealDetails
            deal={deal}
            onClose={this.handleClose}
            open={this.state.open}
          />
        </Hidden>

        
        <Card onClick={this.clickHandler} className={classes.card}>
          <CardActionArea>
            <CardMedia
              style={{ padding: '16px', paddingBottom: '0px' }}
              component="img"
              alt="Contemplative Reptile"
              className={classes.media}
              image={
                deal.picture !== undefined ? deal.picture.url : DefaultImage
              }
              title={deal.title}
            />
            <CardContent>
              <Typography gutterBottom variant="h5" component="h2">
                {deal.title}
              </Typography>
              <div className={classes.flex}>
                <Typography className={classes.shopName}>
                  {deal.shop !== undefined ? deal.shop.title : ''}
                </Typography>
                <Typography>Ends: {deal.end_date}</Typography>
              </div>
            </CardContent>
          </CardActionArea>
        </Card>
      </Fragment>
    );
  }
}

Deal.propTypes = {
  classes: PropTypes.object.isRequired,
  deal: PropTypes.object.isRequired
};

export default withStyles(styles)(Deal);
