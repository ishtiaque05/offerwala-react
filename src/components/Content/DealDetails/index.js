import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';

// import PropTypes from 'prop-types';

const styles = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
    overflow: 'hidden',
    backgroundColor: theme.palette.background.paper
  },
  gridList: {
    width: 500,
    height: 450
  },
  gridItem: {
    marginRight: theme.spacing.unit
  }
});

const DealDetails = ({ classes }) => (
  <div className={classes.root}>
    <Grid container spacing={24}>
      <Grid item xs={12}>
        <Card className={classes.card}>
          <CardActionArea>
            <CardMedia
              component="img"
              alt="Contemplative Reptile"
              className={classes.media}
              height="140"
              image={
                'https://jossdeals-assets-production.s3.amazonaws.com/uploads/deal/picture/1337/Capture.PNG'
              }
              title={'deal.title'}
            />
            <CardContent>
              <Typography gutterBottom variant="h5" component="h2">
                BDT 50 Off on Chillox Burgers
              </Typography>
              <Typography> Lorem, ipsum dolor. </Typography>
              <Typography>
                Will open a detail page, where you î‚ªnd more about the deal, also
                related tags of that deals, and more deals with those related
                tags below, as you can see here this detail says nothing about
                anything related tags of that deals, and more deals with those
                related
              </Typography>
            </CardContent>
          </CardActionArea>
        </Card>
      </Grid>
    </Grid>
  </div>
);

// Shopdeals.propTypes = {
//   classes: PropTypes.object.isRequired,
//   deals: PropTypes.array.isRequired
// };

export default withStyles(styles)(DealDetails);
