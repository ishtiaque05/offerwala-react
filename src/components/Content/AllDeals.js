import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Masonry from 'react-masonry-component';
import InfiniteScroll from 'react-infinite-scroll-component';

import { Typography, withStyles } from '@material-ui/core';

import { fetchAllDeals } from '../../actions';

import Deal from './Deal';

const styles = theme => ({
  root: {
    [theme.breakpoints.up('sm')]: {
      width: theme.spacing.unit * 150,
      marginLeft: theme.spacing.unit * 42
    }
  },
  masonry: {
    width: '100%'
  },
  deal: {
    [theme.breakpoints.down('xs')]: {
      width: '100%'
    }
  }
});

const masonryOptions = {
  transitionDuration: 3
};

class AllDeals extends Component {
  constructor(props) {
    super(props);
    this.state = {
      page: 0,
      deals: []
    };
  }
  componentDidMount() {
    this.props.fetchAllDeals();
  }

  fetchMoreData = () => {
    this.setState({ page: this.state.page + 1 });
    this.props.fetchAllDeals(this.state.page);
    this.setState({ deals: [...this.state.deals, ...this.props.deals] });
  };

  render = () => {
    const { classes, error } = this.props;

    const childElements = this.props.deals.map((deal, index) => (
      <div key={index} className={classes.deal}>
        <Deal deal={deal} />
      </div>
    ));

    const moreElements = this.state.deals.map((deal, index) => (
      <div key={index} className={classes.deal}>
        <Deal deal={deal} />
      </div>
    ));

    if (error) {
      return (
        <div className={classes.root}>
          <Typography variant="body1">Error! {error.message}</Typography>
        </div>
      );
    }

    return (
      <div className={classes.root}>
        <InfiniteScroll
          dataLength={moreElements.length}
          next={this.fetchMoreData}
          hasMore={true}
          loader={<h4>Loading...</h4>}>
          <Masonry
            className={classes.masonry}
            elementType={'div'}
            options={masonryOptions}
            updateOnEachImageLoad={false}>
            {childElements}
            {moreElements}
          </Masonry>
        </InfiniteScroll>
      </div>
    );
  };
}

AllDeals.propTypes = {
  classes: PropTypes.object.isRequired,
  deals: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.object
};

const mapStateToProps = state => ({
  deals: state.deals.items,
  loading: state.deals.loading,
  error: state.deals.error
});

export default connect(
  mapStateToProps,
  { fetchAllDeals }
)(withStyles(styles)(AllDeals));
