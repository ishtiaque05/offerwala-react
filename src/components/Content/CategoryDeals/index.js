import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/core';
import { fetchDealsByCategory } from '../../../actions';
import Deal from '../Deal';
import Masonry from 'react-masonry-component';

const categoryId = {
  'beauty-fitness': 16,
  'food-dining': 8,
  shopping: 9,
  services: 13,
  'mobile-internet': 2,
  'electronics-home': 11,
  'hotel-travels': 14,
  'banks-cards': 12
};

const styles = theme => ({
  root: {
    [theme.breakpoints.up('sm')]: {
      width: '100%'
    },
    [theme.breakpoints.down('lg')]: {
      width: '100%'
    },
    [theme.breakpoints.down('sm')]: {
      width: '98%',
      margin: '0 auto'
    },
    [theme.breakpoints.down('xs')]: {
      width: '96%',
      margin: '0 auto'
    }
  },
  masonry: {
    width: '100%',
    display: 'flex',
    justifyContent: 'space-between'
  }
});

const masonryOptions = {
  transitionDuration: 3
};

class CategoryDealsPage extends Component {
  // this.props.fetchDealsByCategory(categoryId[this.props.match.params.categoryName]);
  // componentDidMount() {
  //   console.log(this.props.match.params.categoryName);
  //   this.props.fetchDealsByCategory(
  //     categoryId[this.props.match.params.categoryName]
  //   );
  // }

  renderDeals() {
    return this.props.deals.map((deal, index) => (
      <React.Fragment key={index}>
        <Deal deal={deal} />
      </React.Fragment>
    ));
  }

  render() {
    const { classes } = this.props;

    return (
      <div className={classes.root}>
        <Masonry
          className={classes.masonry}
          elementType={'div'}
          options={masonryOptions}
          updateOnEachImageLoad={false}>
          {this.renderDeals()}
        </Masonry>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  deals: state.category.items,
  loading: state.category.loading,
  error: state.category.error
});

// export default {
//   loadData: (store, match) =>
//     store.dispatch(fetchDealsByCategory(categoryId[match.params.categoryName])),
//   component: connect(
//     mapStateToProps,
//     { fetchDealsByCategory }
//   )(withStyles(styles)(CategoryDealsPage))
// };

export default connect(
  mapStateToProps,
  { fetchDealsByCategory }
)(withStyles(styles)(CategoryDealsPage));
