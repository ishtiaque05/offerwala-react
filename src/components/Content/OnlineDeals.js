// @flow

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/core/styles';
import GridList from '@material-ui/core/GridList';

import { fetchOnlineDeals } from 'actions';

import Deal from 'components/Content/Deal';
import PropTypes from 'prop-types';

const styles = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
    overflow: 'hidden'
  },
  gridList: {
    width: 500,
    height: 450
  },
  gridItem: {
    marginRight: theme.spacing.unit
  }
});

class OnlineDeals extends Component {
  componentDidMount() {
    this.props.dispatch(fetchOnlineDeals());
  }

  render = () => {
    const { classes, error, loading, deals } = this.props;
    if (error) {
      return <div>Error! {error.message}</div>;
    }

    if (loading) {
      return <div>Loading...</div>;
    }

    return (
      <div className={classes.root}>
        <GridList cellHeight={160} cols={3}>
          {deals.map(deal => (
            <Deal className={classes.gridItem} key={deal.id} deal={deal} />
          ))}
        </GridList>
      </div>
    );
  };
}

OnlineDeals.propTypes = {
  classes: PropTypes.object.isRequired,
  deals: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.object
};

const mapStateToProps = state => ({
  deals: state.deals.items,
  loading: state.deals.loading,
  error: state.deals.error
});

export default connect(mapStateToProps)(withStyles(styles)(OnlineDeals));
