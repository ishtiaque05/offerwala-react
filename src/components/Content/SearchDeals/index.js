import React, { Component } from 'react';
import { connect } from 'react-redux';
import Masonry from 'react-masonry-component';
import { Typography, withStyles } from '@material-ui/core';

import { fetchSearchedDeals } from '../../../actions';
import Deal from '../Deal';

const styles = theme => ({
  root: {
    [theme.breakpoints.up('sm')]: {
      // width: theme.spacing.unit * 150
      width: '100%'
    },
    [theme.breakpoints.down('lg')]: {
      // width: theme.spacing.unit * 100
      width: '100%'
    },
    [theme.breakpoints.down('sm')]: {
      width: '98%',
      margin: '0 auto'
    }, [theme.breakpoints.down('xs')]: {
      width: '96%',
      margin: '0 auto', 
      
    }
  },
  masonry: {
    width: '100%', 
    display: 'flex', 
    justifyContent: 'space-between'
  }
});

const masonryOptions = {
  transitionDuration: 3
};

class SearchDeals extends Component {
  // state = {
  //   result = null
  // }

  componentDidMount() {
    console.warn(this.props.match.params.tagName);
    this.props.fetchSearchedDeals(this.props.match.params.tagName);
    
  }

  render() {
    
    const { classes, loading, error, deals } = this.props;
    // console.log(deals);

    let childElements = null;

    if (loading) {
      return (
        <div className={classes.root}>
          <Typography variant="body1">Loading...</Typography>
        </div>
      );
    } 

    if(!deals) {
      console.log(deals);
      childElements = (
        <div className={classes.root}>
          <Typography variant="body1">No Deals Found</Typography>
        </div>
      )
    } 

    if(deals) {
      console.log(deals);
      childElements = deals.map((deal, index) => (
        <React.Fragment key={index}>
          <Deal deal={deal} />
        </React.Fragment>
      ));
    }

    if (error) {
      return (
        <div className={classes.root}>
          <Typography variant="body1">Error! {error.message}</Typography>
        </div>
      );
    }

    return (
      <div className={classes.root}>
        <Masonry
          className={classes.masonry}
          elementType={'div'}
          options={masonryOptions}
          updateOnEachImageLoad={false}>
          {childElements}
        </Masonry>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  deals: state.searchedDeals.items,
  loading: state.searchedDeals.loading,
  error: state.searchedDeals.error
});

export default connect(
  mapStateToProps,
  { fetchSearchedDeals }
)(withStyles(styles)(SearchDeals));
