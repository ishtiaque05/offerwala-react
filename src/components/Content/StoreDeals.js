import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Masonry from 'react-masonry-component';
import InfiniteScroll from 'react-infinite-scroll-component';

import { withStyles, Typography } from '@material-ui/core';

import { fetchStoreDeals } from '../../actions';

import Deal from './Deal';

const styles = theme => ({
  root: {
    [theme.breakpoints.up('sm')]: {
      width: theme.spacing.unit * 150
    },
    [theme.breakpoints.down('lg')]: {
      width: theme.spacing.unit * 100
    },
    [theme.breakpoints.down('sm')]: {
      width: '98%',
      margin: '0 auto'
    }, [theme.breakpoints.down('xs')]: {
      width: '96%',
      margin: '0 auto', 
      
    }
  },
  masonry: {
    width: '100%'
  },
  deal: {
    [theme.breakpoints.down('xs')]: {
      width: '100%'
    }
  }
});

const masonryOptions = { transitionDuration: 3 };

class StoreDeals extends Component {
  constructor(props) {
    super(props);
    this.state = {
      page: 0,
      deals: []
    };
  }

  componentDidMount() {
    this.props.fetchStoreDeals();
  }

  fetchMoreData = () => {
    this.setState({ page: this.state.page + 1 });
    this.props.fetchStoreDeals(this.state.page);
    this.setState({ deals: [...this.state.deals, ...this.props.deals] });
  };

  render = () => {
    const { classes, error } = this.props;

    const childElements = this.props.deals.map((deal, index) => (
      <React.Fragment key={index}>
        <Deal deal={deal} />
      </React.Fragment>
    ));

    const moreElements = this.state.deals.map((deal, index) => (
      <React.Fragment key={index}>
        <Deal deal={deal} />
      </React.Fragment>
    ));

    if (error) {
      return (
        <div className={classes.root}>
          <Typography variant="body1">Error! {error.message}</Typography>
        </div>
      );
    }

    return (
      <div className={classes.root}>
        <InfiniteScroll
          dataLength={moreElements.length}
          next={this.fetchMoreData}
          hasMore={true}
          loader={<h4>Loading...</h4>}>
          <Masonry
            className={classes.masonry}
            elementType={'div'}
            options={masonryOptions}
            updateOnEachImageLoad={false}>
            {childElements}
            {moreElements}
          </Masonry>
        </InfiniteScroll>
      </div>
    );
  };
}

StoreDeals.propTypes = {
  classes: PropTypes.object.isRequired,
  deals: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.object
};

const mapStateToProps = state => ({
  deals: state.deals.items,
  loading: state.deals.loading,
  error: state.deals.error
});

export default connect(
  mapStateToProps,
  { fetchStoreDeals }
)(withStyles(styles)(StoreDeals));
