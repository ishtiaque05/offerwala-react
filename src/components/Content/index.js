import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/core/styles';
import GridList from '@material-ui/core/GridList';

import { fetchDeals } from 'actions';

import Deal from 'components/Content/Deal';
import PropTypes from 'prop-types';

const styles = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
    overflow: 'hidden',
    backgroundColor: theme.palette.background.paper
  },
  gridList: {
    width: 500,
    height: 450
  },
  gridItem: {
    marginRight: theme.spacing.unit
  }
});

class Content extends Component {
  componentDidMount() {
    this.props.dispatch(fetchDeals());
  }

  render = () => {
    const { classes, error, loading, deals } = this.props;
    if (error) {
      return <div>Error! {error.message}</div>;
    }

    if (loading) {
      return <div>Loading...</div>;
    }

    return (
      <div className={classes.root}>
        <GridList cellHeight={160} cols={3}>
          {deals.map(deal => (
            <Deal className={classes.gridItem} key={deal.id} deal={deal} />
          ))}
        </GridList>
      </div>
    );
  };
}

Content.propTypes = {
  classes: PropTypes.object.isRequired,
  deals: PropTypes.array.isRequired
};

const mapStateToProps = state => ({
  deals: state.deals.items,
  loading: state.deals.loading,
  error: state.deals.error
});

export default connect(mapStateToProps)(withStyles(styles)(Content));
