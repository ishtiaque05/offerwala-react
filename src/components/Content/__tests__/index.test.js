import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Content from 'components/Content';

describe('Content', () => {
  let component;
  let testProps;

  beforeEach(() => {
    testProps = {
      deals: [
        {
          imgUrl: 'default.jpg',
          title: 'Default',
          validUntil: '02 Nov',
          storeTitle: 'Default'
        }
      ]
    };
    component = shallow(<Content {...testProps} />);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Content {...testProps} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
