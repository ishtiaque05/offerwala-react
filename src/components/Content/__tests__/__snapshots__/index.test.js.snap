// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<div
  className="Content-root-1"
>
  <ul
    className="MuiGridList-root-4"
    style={
      Object {
        "margin": -2,
      }
    }
  >
    <div
      className="MuiPaper-root-7 MuiPaper-elevation1-10 MuiPaper-rounded-8 MuiCard-root-6 Deal-card-5"
    >
      <button
        className="MuiButtonBase-root-37 MuiCardActionArea-root-34"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <img
          alt="Contemplative Reptile"
          className="MuiCardMedia-root-40 MuiCardMedia-media-41"
          height="140"
          src="default.jpg"
          title="Default"
        />
        <div
          className="MuiCardContent-root-42"
        >
          <h2
            className="MuiTypography-root-43 MuiTypography-h5-59 MuiTypography-gutterBottom-70"
          >
            Default
          </h2>
          <p
            className="MuiTypography-root-43 MuiTypography-body1-52"
          />
          <p
            className="MuiTypography-root-43 MuiTypography-body1-52"
          >
            02 Nov
          </p>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight-36"
        />
        <span
          className="MuiTouchRipple-root-78"
        />
      </button>
    </div>
  </ul>
</div>
`;
